test/test_icmpv4.ml

```diff
@@ -92,10 +92,13 @@ let echo_request () =
   inform_arp listener speaker_address (mac_of_stack speaker);
   let req = Icmpv4_packet.({code = 0x00; ty = Icmpv4_wire.Echo_request;
                             subheader = Id_and_seq (id_no, seq_no)}) in
-  let echo_request = Icmpv4_packet.Marshal.make_cstruct req ~payload:request_payload in
+  let echo_request = Cstruct.create 2048 in
+  Icmpv4_packet.Marshal.into_cstruct req echo_request ~payload:request_payload >>=? fun () ->
+  Cstruct.blit request_payload 0 echo_request (Icmpv4_wire.sizeof_icmpv4) (Cstruct.len request_payload);
+  let echo_request = Cstruct.sub echo_request 0 (Icmpv4_wire.sizeof_icmpv4 + Cstruct.len request_payload) in
```

Comment:
I don't quite understand the necessity for this hunk. the previous `make_cstruct` looks cleaner to me

Reply:
Motivation was trying to get some test coverage for `Icmpv4_packet.Marshal.make_cstruct`, to avoid future regressions.  It is less clear, unfortunately.